neuroanalysis/synaptic_release.py

```diff
@@ -169,10 +231,114 @@ def run_fit(self, spike_sets):
         
         Returns a dictionary of optimal model parameters.        
         """
-        init = [self.dict_params[param] for param in self.order]
-        bounds = [self.dict_bounds[param] for param in self.order]
+        #print spike_sets
+        self.data_x=[]
+        self.data_y=[]
+        self.data_e=[]
+        lengths=[]
+        for datax, datay in spike_sets:
```

Comment:
I'm not following what's happening here-- we start with a structure of nested lists in `spike_sets`, then flatten it into data_x, data_y, and lengths, and then reconstruct the same nested structure in feval?

Reply:
You are right. That is what it is. I had to do this weird trick to use ‘model’ class in lmfit. You suggested the way to pass spike_sets to the fitting routine. It worked well until I started using the model class. 
But when I started using the model class, lmfit threw out the error message suggesting that the number of data points is not sufficient to estimate the parameters in the synapse model. In my impression, lmfit considers the length of spike_sets as the number of data, and it complains about the lack of data points. To avoid this issue, I made spike_sets into a long array and later reconstructed them in the original shape. I did not think it was an elegant solution, but it worked.

Reply:
Got it, thanks.